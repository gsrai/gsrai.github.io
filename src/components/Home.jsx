import React from 'react';
import { Route } from 'react-router-dom';

import Header from './Header.jsx';
import Footer from './Footer.jsx';
import ProjectsContainer from './ProjectsContainer.jsx';
import About from './About.jsx';
import Blog from './Blog.jsx';

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      projectsModel: {
        projects: [
          {
            title: 'Tab Manager Pro',
            desc: 'Personal project to learn the chrome extension api and solidify my React and Redux knowledge. It is available on the chrome store and the code on github.',
            imgsrc: 'assets/tabmanagerpro.png',
            links: [
              {
                href: 'https://github.com/gsrai/tab-manager-pro',
                title: 'Github'
              },
              {
                href: 'https://chrome.google.com/webstore/detail/tab-manager-pro/ebfpahimijhihjjnepkoojfpgpllbcdf',
                title: 'Chrome Store'
              }
            ]
          },
          {
            title: 'Word Envy',
            desc: 'A puzzle game that I created to learn React Native and publish an app on both the google play store and the apple app store.',
            imgsrc: 'assets/wordenvy.jpg',
            links: [
              {
                href: 'https://play.google.com/store/apps/details?id=com.wordenvy',
                title: 'Google Play'
              },
              {
                href: 'https://itunes.apple.com/us/app/word-envy/id1266290411',
                title: 'App Store'
              }
            ]
          },
          {
            title: 'JS game clones',
            desc: 'I made Snake and Pong game clones written in es5 Javascript, I wanted to make games and learn the HTML5 canvas API.',
            imgsrc: 'assets/pong.png',
            links: [
              {
                href: 'https://github.com/gsrai/portfolio/tree/master/Pong',
                title: 'Pong'
              },
              {
                href: 'https://github.com/gsrai/portfolio/tree/master/Snake',
                title: 'Snake'
              }
            ]
          },
          {
            title: 'Personal Website and Blog',
            desc: 'I used a github site to host my own website which would present a portfolio of my work and a blog written in Javascript and uses React.',
            imgsrc: 'assets/portfolio.png',
            links: [
              {
                href: 'https://github.com/gsrai/gsrai.github.io',
                title: 'Github'
              }
            ]
          }
        ]
      },
      aboutModel: {
        name: 'Gagondeep Srai',
        age: '23 Years old',
        occupation: 'Software Developer',
        location: 'London, UK'
      }
    };
  }

  render() {
    const { projectsModel, aboutModel }  = this.state;
    return (
      <div>
        <Header />
        <Route exact={true} path="/" render={() => {
          return (<ProjectsContainer projectsModel={projectsModel} />);
        }}></Route>
        <Route exact path="/about" render={() => {
          return (<About {...aboutModel} />);
        }}></Route>
        <Route exact path="/blog" component={Blog}></Route>
        <Footer />
      </div>
    );
  }
}