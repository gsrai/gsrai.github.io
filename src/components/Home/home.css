body {
  background-color: whitesmoke;
  margin: 1rem;
}

.to-top {
  z-index: 100;
  opacity: 1;
  display: inline-block;
  padding: 1em;
  position: fixed;
  bottom: 1em;
  right: 1em
}

.to-top .paper-btn {
  z-index: 999;
  padding: .6em 1em;
  background: #fff;
  border-top-left-radius: 185px 160px;
  border-top-right-radius: 200px 195px;
  border-bottom-right-radius: 160px 195px;
  border-bottom-left-radius: 185px 190px;
}

@media only screen and (max-width : 420px) {
  body {
    margin: 0;
  }
}