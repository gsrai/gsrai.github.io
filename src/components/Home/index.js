import React from 'react'
import { Route } from 'react-router-dom'
import { hot } from 'react-hot-loader'

import Header from '../Header'
import Footer from '../Footer'
import ProjectsContainer from '../ProjectsContainer'
import About from '../About'
import Blog from '../Blog'
import ErrorBoundary from '../../utils/ErrorBoundary'

import './home.css'
import BlogPostView from '../BlogPostView'

class Home extends React.Component {
  constructor(props) {
    super(props)
    const now = new Date()
    const dob = new Date(775699200000)
    const age = now.getFullYear() - dob.getFullYear()

    this.state = {
      projectsModel: {
        projects: [
          {
            title: 'Tab Manager Pro',
            desc: 'Personal project to learn the chrome extension api and solidify my React and Redux knowledge. It is available on the chrome store and the code on github.',
            imgsrc: 'assets/tabmanagerpro.png',
            links: [
              {
                href: 'https://github.com/gsrai/tab-manager-pro',
                title: 'Github'
              },
              {
                href: 'https://chrome.google.com/webstore/detail/tab-manager-pro/ebfpahimijhihjjnepkoojfpgpllbcdf',
                title: 'Chrome Store'
              }
            ]
          },
          {
            title: 'Word Envy',
            desc: 'A puzzle game that I created to learn React Native and publish an app on the google play store.',
            imgsrc: 'assets/wordenvy.jpg',
            links: [
              {
                href: 'https://play.google.com/store/apps/details?id=com.wordenvy',
                title: 'Google Play'
              }
            ]
          },
          {
            title: 'JS game clones',
            desc: 'I made Snake and Pong game clones written in es5 Javascript, I wanted to make games and learn the HTML5 canvas API.',
            imgsrc: 'assets/pong.png',
            links: [
              {
                href: 'https://github.com/gsrai/portfolio/tree/master/Pong',
                title: 'Pong'
              },
              {
                href: 'https://github.com/gsrai/portfolio/tree/master/Snake',
                title: 'Snake'
              }
            ]
          },
          {
            title: 'Personal Website and Blog',
            desc: 'I used a github site to host my own website which would present a portfolio of my work and a blog written in Javascript and uses React.',
            imgsrc: 'assets/portfolio.png',
            links: [
              {
                href: 'https://github.com/gsrai/gsrai.github.io',
                title: 'Github'
              }
            ]
          }
        ]
      },
      aboutModel: {
        name: 'Gagondeep Srai',
        age,
        occupation: 'Software Developer',
        location: 'London, UK'
      }
    }
  }

  scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    })
  }

  render() {
    const { projectsModel, aboutModel } = this.state
    return (
      <div className='paper'>
        <ErrorBoundary>
          <Header />
          <div className='to-top'>
            <button className='paper-btn margin' onClick={this.scrollToTop}>^</button>
          </div>
          <Route exact path='/' render={() => {
            return (<ProjectsContainer projectsModel={projectsModel} />)
          }} />
          <Route exact path='/about' render={() => {
            return (<About {...aboutModel} />)
          }} />
          <Route exact path='/blog' component={Blog} />
          <Route path='/blog/posts/*' component={BlogPostView} />
          <Footer />
        </ErrorBoundary>
      </div>
    )
  }
}

export default hot(module)(Home)
