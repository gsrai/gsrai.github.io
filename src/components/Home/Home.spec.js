import { shallow } from 'enzyme'
import React from 'react'
import Home from './index'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(<Home />)
    expect(wrapper).toMatchSnapshot()
  })
})
