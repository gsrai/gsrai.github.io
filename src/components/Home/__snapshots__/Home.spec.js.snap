// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div
  className="paper"
>
  <ErrorBoundary>
    <Header />
    <div
      className="to-top"
    >
      <button
        className="paper-btn margin"
        onClick={[Function]}
      >
        ^
      </button>
    </div>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/about"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/blog"
    />
    <Route
      component={[Function]}
      path="/blog/posts/*"
    />
    <Footer />
  </ErrorBoundary>
</div>
`;
